### PR TITLE
feat(bspline): add multiply and __mul__ for exact 1D B-spline product

### PR DESCRIPTION
**Pull request by pantolin**
## Summary

- Implements exact pointwise multiplication of two 1D B-splines via full-Bézier extraction and the Bernstein product formula
- Works for both non-rational and rational (NURBS) operands; non-rational is promoted to rational with unit weights when needed
- New `_bspline_product.py` (Layer 2) + `Bspline.multiply` / `Bspline.__mul__` on the public API

## Algorithm

1. Gather interior breakpoints and multiplicities from both spaces
2. Merge into a union mesh via two-pointer scan
3. Refine each operand to full-Bézier (C⁰ at every interior breakpoint) by knot insertion
4. Apply the Bernstein product formula element-by-element
5. Assemble control points into a full-Bézier product space of degree `p + q`

The product knot vector uses interior multiplicity `p + q` (full-Bézier), giving `n_elements * (p+q) + 1` basis functions — consistent with the assembled control points.

**Note:** The plan described `sum_mults = m_f + m_g` as the interior multiplicity of the product knot vector, but that is inconsistent with the assembled control point count. The correct representation is full-Bézier (interior mult = `p + q`), which is always a valid (though not minimal) representation of the exact product.

## Test plan

- [x] Non-rational: same knot vector, different knot vectors, different degrees, additive multiplicity, vector rank, three-element mesh, `__mul__` operator
- [x] Rational: non-rational × rational (mixed promotion), rational × rational with non-unit weights
- [x] Edge cases: single Bézier element, multiply by constant, multiply by 1
- [x] Error paths: `dim ≠ 1`, dtype mismatch, rank mismatch, domain mismatch, periodic splines
- [x] All correctness tests evaluate at 201 points and assert `allclose` with `atol=1e-11`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/pantr/_bspline_product.py [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+    dtype = bp1.dtype
+    if len(all_bp_list) == 0:
+        return np.empty(0, dtype=dtype), np.empty(0, dtype=np.int_)
+    return (
+        np.array(all_bp_list, dtype=dtype),
+        np.array(sum_mults_list, dtype=np.int_),
+    )
+
+
+def _lookup_mults_in_space(
```
Comment (pantolin) [inline src/pantr/_bspline_product.py:117]: @claude look for existing function doing something similar?

Comment (pantolin) [reply src/pantr/_bspline_product.py:117]: No dedicated existing function for this specific lookup, but the codebase already uses `np.searchsorted` for sorted-array lookups in `_get_last_knot_smaller_equal_impl` and `_compute_oslo_matrix_1d_core`. Replaced the manual two-pointer scan with a vectorised `searchsorted` approach — simpler and consistent with the existing pattern.
